### PR TITLE
feature/frontend rbac and admin creds

### PR DESCRIPTION
**Pull request by msadovyi**

### CODE REVIEW

File: packages/backend/config/env/development.json [hunk 1]
```diff
@@ -12,6 +12,10 @@
       "confirmationTemplateId": "@@SENDGRID_CONFIRMATION_TEMPLATE",
       "resetPasswordTemplateId": "@@SENDGRID_RESET_PASSWORD_TEMPLATE",
       "apiKey": "@@SENDGRID_API_KEY"
+    },
+    "defaultAdmin": {
```
Comment (valcker) [inline packages/backend/config/env/development.json:16]: Please update documentation, docker files, docker-compose and examples to be in line with these changes.

File: packages/backend/src/modules/user/user.service.ts [hunk 2]
```diff
@@ -113,15 +114,17 @@ export default class UserService {
   }
 
   public async createDefaultUser(): Promise<void> {
-    const toCreate = await this.userModel.findOne({
-      email: config.application.defaultAdmin.email,
-    });
+    const { email, password } = config.application.defaultAdmin;
+
+    if (email === 'admin@example.com' || password === 'Secret123!') {
```
Comment (valcker) [inline packages/backend/src/modules/user/user.service.ts:119]: Do we still have default credentials? I see that we replaced them with @@ variables in config files.

File: packages/frontend/src/presenter/DataSources/index.ts [hunk 3]
```diff
@@ -19,4 +19,4 @@ export { useSoapHandler } from './useSoapHandler';
 export { useEditorsHandler } from './useEditorsHandler';
 export { useTransforms } from './useTransforms';
 export { useTransformEditors } from './useTransformEditors';
-export { useUpdateDataSource } from './useUpdateDataSource';
+export { useUpdateDataSource } from './useUpdateDataSource';
```
Comment (valcker) [inline packages/frontend/src/presenter/DataSources/index.ts:22]: No newline at the end of the file. Prettier?

File: packages/frontend/src/view/Navigation/Main/constants.ts [hunk 4]
```diff
@@ -50,21 +55,25 @@ export const GROUPS = [
...
         text: 'Webhooks',
         link: ROUTES.WEBHOOKS,
         Icon: AllInclusive,
+        roles: [Roles.ADMIN, Roles.USER],
       },
       {
         text: 'Gateway nodes',
         link: ROUTES.NODES,
         Icon: Ballot,
+        roles: [Roles.ADMIN, Roles.USER],
```
Comment (valcker) [inline packages/frontend/src/view/Navigation/Main/constants.ts:76]: Admin only

File: packages/frontend/src/view/RoleProtectedComponent/RoleProtectedComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+export const RoleProtectedComponent = <T,>(roles: Roles[], Component: React.FC<T>, redirectTo?: string): React.FC<T> => {
+  const Wrapped: React.FC<T> = (props) => {
+    const { role } = useAuth();
+    if (roles.includes(role)) return (<Component {...props} />);
+    if (redirectTo) return (<Redirect to={redirectTo} />)
+    return <></>;
+  }
+
+  return Wrapped;
+}
```
Comment (valcker) [inline packages/frontend/src/view/RoleProtectedComponent/RoleProtectedComponent.tsx:14]: no new line at the end of the file. prettier?

**Review by valcker — APPROVED** [review]
Please fix the comments first.

File: packages/backend/src/modules/user/user.service.ts (hunk 2 above)
Comment (msadovyi) [reply packages/backend/src/modules/user/user.service.ts:119]: Yes, we still have default credentials in config/default.json

**Pull request merged**